src/redux/explorer.js

```diff
@@ -422,26 +420,57 @@ export const explorerInitL2 = (network = null) => (dispatch, getState) => {
   if (getState().settings.network === NetworkTypes.mainnet) {
     switch (network) {
       case NetworkTypes.arbitrum:
-        // Start watching arbitrum assets
-        dispatch(arbitrumExplorerInit());
+      case NetworkTypes.polygon:
+        // Start watching arbitrum and polygon assets
+        dispatch(fetchAssetsFromRefraction());
         break;
       case NetworkTypes.optimism:
         // Start watching optimism assets
         dispatch(optimismExplorerInit());
         break;
-      case NetworkTypes.polygon:
-        // Start watching polygon assets
-        dispatch(polygonExplorerInit());
-        break;
       default:
         // Start watching all L2 assets
-        dispatch(arbitrumExplorerInit());
         dispatch(optimismExplorerInit());
-        dispatch(polygonExplorerInit());
     }
   }
 };
 
+const fetchAssetsFromRefraction = () => (dispatch, getState) => {
+  const { accountAddress, nativeCurrency } = getState().settings;
+  const { addressSocket } = getState().explorer;
+  addressSocket.emit([
+    'get',
+    {
+      payload: {
+        address: accountAddress,
+        currency: toLower(nativeCurrency),
+      },
+      scope: ['assets'],
+    },
+  ]);
+};
+
+const l2AddressAssetsReceived = (message, network) => (dispatch, getState) => {
+  const { assets: allAssets, genericAssets } = getState().data;
+
+  const updatedMessage = Object.assign({}, message);
+  updatedMessage.payload.assets = message?.payload?.assets?.map(asset => {
+    const fallbackAsset =
+      ethereumUtils.getAsset(allAssets, toLower(asset.asset.mainnet_address)) ||
+      genericAssets[toLower(asset.asset.mainnet_address)];
+
+    if (fallbackAsset) {
+      asset.asset.price = {
+        ...asset.asset.price,
+        ...fallbackAsset.price,
+      };
+    }
+    return asset;
+  });
+
+  dispatch(addressAssetsReceived(updatedMessage, network));
```

Comment:
second param on `addressAssetsReceived` is not the network
think you're looking for `addressAssetsReceived(updatedMessage, false, false, false, network)`

Reply:
Believe this is causing:
![IMG_0053](https://user-images.githubusercontent.com/1285228/144733533-6f44e320-828a-40eb-b2e9-901bf6c498f2.PNG)